Ex1.java

```diff
@@ -0,0 +1,134 @@
+import java.util.HashMap;
+
+public class Ex1 {
+
+    /**
+     * The function receives two strings and checks if one is a permutation
+     * of the other
+     * @param firstStr
+     * @param secondStr
+     * @return true if one is a permutation of the other, otherwise return false
+     */
+    public static boolean isPermutation (String firstStr, String secondStr){
+        // Check if one of the strings are null or if their length is not equal
+        if (firstStr == null || secondStr == null || firstStr.length() != secondStr.length()){
+            return false;
+        }
+
+        HashMap<Character, Integer> map = new HashMap<Character, Integer>();
+
+        // Create a hash map so that the key is the character
+        // and the value is the number of times that the character appears in the string
+        for(int i = 0; i < firstStr.length(); i++){

```

Comment:
It may seem not so important, but sticking to established coding conventions makes reading code much easier. Java [coding conventions](http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141388.html#682) explicitly mention putting space between keywords like 'for' and opening brace.

Typically one would add space also before { that opens the block. Could you please add spaces where necessary?